src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -181,6 +181,7 @@ export class RecommendationHandler {
         let shouldRecordServiceInvocation = true
         session.language = runtimeLanguageContext.getLanguageContext(editor.document.languageId).language
         session.taskType = await this.getTaskTypeFromEditorFileName(editor.document.fileName)
+        session.requestIdList = []
```

Comment:
don't think this place is correct. This method will be called multiple times with paginated call. If the requestIdList field serves to store the request ids within a session, it will reset every API call.

Reply:
should move to clearRecommendation() instead.

Reply:
emm yes, how about on like 212?

Reply:
actually clearRecommendation() is better.